Scripts/Runtime/Data/IDProviderAtomicBurstable.cs

```diff
@@ -0,0 +1,76 @@
+using Anvil.CSharp.Data;
+using System.Threading;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using UnityEngine;
+
+namespace Anvil.Unity.DOTS.Data
+{
+    /// <summary>
+    /// An <see cref="IDProvider"/> that is compatible with Burst and will get the next ID atomically so it is
+    /// safe to use in threaded context.
+    /// </summary>
+    [BurstCompatible]
+    public readonly unsafe struct IDProviderAtomicBurstable
```

Comment:
I don't think there's currently a use case but should we have a TODO to make this generic like `AbstractIDProvider<T>`?

Reply:
I've added a far future TODO that relies on C#11 so we can have generic math via the static interfaces. Since this is a struct we can't inherit and there's no way to do anything with the `T` part yet.